includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2229,31 +2230,37 @@ private function deserialize_charge_object_from_array( array $charge_array ) {
 	}
 
 	/**
-	 * De-serialize an intention array into a intention object
+	 * De-serialize an intention array into a payment intention object
 	 *
 	 * @param array $intention_array - The intention array to de-serialize.
 	 *
-	 * @return WC_Payments_API_Intention
+	 * @return WC_Payments_API_Payment_Intention
 	 * @throws API_Exception - Unable to deserialize intention array.
 	 */
-	public function deserialize_intention_object_from_array( array $intention_array ) {
+	public function deserialize_payment_intention_object_from_array( array $intention_array ) {
 		// TODO: Throw an exception if the response array doesn't contain mandatory properties.
 		$created = new DateTime();
 		$created->setTimestamp( $intention_array['created'] );
 
-		$charge_array         = 0 < $intention_array['charges']['total_count'] ? end( $intention_array['charges']['data'] ) : null;
-		$next_action          = ! empty( $intention_array['next_action'] ) ? $intention_array['next_action'] : [];
-		$last_payment_error   = ! empty( $intention_array['last_payment_error'] ) ? $intention_array['last_payment_error'] : [];
-		$metadata             = ! empty( $intention_array['metadata'] ) ? $intention_array['metadata'] : [];
-		$customer             = $intention_array['customer'] ?? $charge_array['customer'] ?? null;
-		$payment_method       = $intention_array['payment_method'] ?? $intention_array['source'] ?? null;
-		$processing           = $intention_array[ Payment_Intent_Status::PROCESSING ] ?? [];
-		$payment_method_types = $intention_array['payment_method_types'] ?? [];
+		// Metadata can be an empty stdClass object, so we need to check array type too.
+		// See https://github.com/Automattic/woocommerce-payments/pull/419/commits/c2c8438c3ed7be6d604435e059209fb87fb6d0c4.
+		$raw_metadata           = $intention_array['metadata'];
+		$metadata               = is_array( $raw_metadata ) && ! empty( $raw_metadata )
+			? $raw_metadata
+			: [];
+		$charge_array           = 0 < $intention_array['charges']['total_count'] ? end( $intention_array['charges']['data'] ) : null;
+		$next_action            = ! empty( $intention_array['next_action'] ) ? $intention_array['next_action'] : [];
```

Comment:
Can you use `$intention_array['next_action'] ?? []` here? It is a small nitpick but makes the code a bit easier to read.

Reply:
Addressed fe80244